### PR TITLE
Add telemetry dependency and configuration files.

### PR DESCRIPTION
**Pull request by Incarnation-p-lee**
Signed-off-by: Pan Li <panli@microsoft.com>

### CODE REVIEW

File: spring-cloud-azure-autoconfigure/src/main/resources/META-INF/project.properties [hunk 1]
```diff
@@ -0,0 +1 @@
+project.version=@project.version@
```
Comment (warrenzhu25) [inline spring-cloud-azure-autoconfigure/src/main/resources/META-INF/project.properties]: Why this properties is inside META-INF?

File: spring-cloud-azure-autoconfigure/src/main/resources/ApplicationInsights.xml [hunk 2]
```diff
@@ -0,0 +1,12 @@
+<?xml version="1.0" encoding="utf-8"?>
+<ApplicationInsights xmlns="http://schemas.microsoft.com/ApplicationInsights/2013/Settings" schemaVersion="2014-05-30">
+    <!-- The key from the portal: -->
```
Comment (warrenzhu25) [inline spring-cloud-azure-autoconfigure/src/main/resources/ApplicationInsights.xml]: I don't see any useful information in this xml file. Is this file required?

File: spring-cloud-azure-autoconfigure/pom.xml [hunk 3]
```diff
@@ -73,6 +77,34 @@
...
+
     </dependencies>
 
-</project>
\ No newline at end of file
+    <build>
+        <resources>
+            <!-- Filtering Resource -->
+            <resource>
+                <directory>src/main/resources</directory>
```
Comment (warrenzhu25) [inline spring-cloud-azure-autoconfigure/pom.xml]: This filter means not including this file in building phase, right? Then why need this file?

Comment (Incarnation-p-lee) [reply spring-cloud-azure-autoconfigure/pom.xml]: No, it will include the file and replace the `placeholder`.

File: spring-cloud-azure-autoconfigure/src/main/resources/META-INF/project.properties (hunk 1 above)
Comment (Incarnation-p-lee) [reply spring-cloud-azure-autoconfigure/src/main/resources/META-INF/project.properties]: From convention here, telemetry will take this version as property.

File: spring-cloud-azure-autoconfigure/src/main/resources/ApplicationInsights.xml (hunk 2 above)
Comment (Incarnation-p-lee) [reply spring-cloud-azure-autoconfigure/src/main/resources/ApplicationInsights.xml]: From AI SDK will use this, and only fill the Key when release.
